### PR TITLE
fix(Tile): update TS structure

### PR DESCRIPTION
**Pull request by arwehrman**
## Description

This PR updates Tile type declarations

## References

LUI-843

## Checklist

- [ ] all commented code has been removed
- [ ] any new console issues have been resolved
- [ ] code linter and formatter has been run
- [ ] test coverage meets repo requirements
- [ ] PR name matches the expected semantic-commit syntax

### CODE REVIEW

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts [hunk 1]
```diff
@@ -36,21 +38,91 @@ export type TileStyle = SurfaceStyle & {
...
-export default class Tile extends Surface {
-  // TODO: Replace value with PatchTemplate<typeof Artwork> when the Component template specs are created
-  artwork?: Record<string, unknown>;
+declare namespace Tile {
+  export interface TemplateSpec extends Surface.TemplateSpec {
+    Content: typeof lng.Component<lng.Component.TemplateSpecLoose>;
+    /**
+     * Object containing all properties supported in the [Artwork component](?path=/docs/components-artwork--artwork)
+     */
+    artwork: lightning.Element.PatchTemplate<Artwork.TemplateSpec>;
```
Comment (arwehrman) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: I believe this how we are supposed to do these, given that `updateArtwork` is patching to Artwork

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts [hunk 2]
```diff
@@ -18,10 +18,20 @@
 
 import lng from '@lightningjs/core';
 import { MetadataBaseStyle } from '../MetadataBase';
```
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: Just drawing your attention to the warning that this should be removed.

**Review by ImCoolNowRight — APPROVED** [review]
Just needs an import removed, but otherwise looks good!

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts (hunk 2 above)
Comment (anthony9187) [reply packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: but are we sure it's never used? /s

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts [hunk 3]
```diff
@@ -18,10 +18,20 @@
...
+import {
+  Artwork,
+  TextBox,
+  ProgressBar,
+  MetadataBase,
+  Badge,
+  Checkbox,
+  Label,
+  Surface
+} from '..';
```
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: as a rule we're not using intermediate imports in LUI, these should each be imported separately 😬 sorry

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts [hunk 4]
```diff
@@ -18,10 +18,20 @@
...
 
-export type TileStyle = SurfaceStyle & {
+type TileStyle = SurfaceStyle & {
   animationEntrance: Record<string, unknown>;
```
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: I believe core has a transition type we're leveraging for this?

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts [hunk 5]
```diff
@@ -31,33 +41,103 @@ export type TileStyle = SurfaceStyle & {
...
+     */
+    circle?: boolean;
+    /**
+     * Object containing all properties supported in the [Label component](?path=/docs/components-label--label)
+     */
+    label?: lng.Element.PatchTemplate<Label.TemplateSpec>;
+    /**
+     * Controls where there metadata is displayed in relation to the Tile. Available values are 'standard' and 'inset'
+     */
+    metadataLocation?: string;
```
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: we can make this stricter - `'standard' | 'inset'`

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts [hunk 6]
```diff
@@ -31,33 +41,103 @@ export type TileStyle = SurfaceStyle & {
...
+   */
+  circle?: boolean;
+  /**
+   * Object containing all properties supported in the [Label component](?path=/docs/components-label--label)
+   */
+  label?: lng.Element.PatchTemplate<Label.TemplateSpec>;
+  /**
+   * Controls where there metadata is displayed in relation to the Tile. Available values are 'standard' and 'inset'
+   */
   metadataLocation?: string;
```
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: same as above

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts (hunk 2 above)
Comment (anthony9187) [reply packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: @arwehrman why do we need MetadataBaseStyle to be imported? if it's not being used by anything we should remove the import

Comment (arwehrman) [reply packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: removed

**Comment by soumyaloka** [comment]
can we add a signalmap for imageLoaded? Otherwise this looks good!

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts [hunk 7]
```diff
@@ -17,47 +17,124 @@
...
+import { StylePartial } from '../../types/lui';
+import { SurfaceStyle } from '../Surface';
+import Artwork from '../Artwork';
+import TextBox from '../TextBox';
+import ProgressBar from '../ProgressBar';
+import MetadataBase from '../MetadataBase';
+import Badge from '../Badge';
+import Checkbox from '../Checkbox';
+import Label from '../Label';
+import Surface from '../Surface';
```
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.d.ts]: can we combine this with the SurfaceStyle import on line 21?

**Review by anthony9187 — APPROVED** [review]

**Pull request merged**